srcpkgs/xen/files/xenstored/run

```diff
@@ -1,7 +1,26 @@
 #!/bin/sh
+exec 2>&1
 [ ! -d /run/xen ] && mkdir -p /run/xen
-modprobe -q xen-evtchn xen-gnttalloc || exit 1
 mountpoint -q /proc/xen || mount -t xenfs xenfs /proc/xen
+# If not running in a Xen dom0, then this service is a no-op
+# and we can't start a real xenstored.
+grep -q control_d /proc/xen/capabilities || \
+    exec chpst -b \
+	 'shim-xenstored; see /etc/sv/xenstored/run to find out what this is' \
+	 pause
 mountpoint -q /var/lib/xenstored || mount -t tmpfs xenstored /var/lib/xenstored
-grep -q control_d /proc/xen/capabilities || exit 1
-exec xenstored --verbose --no-fork
+
+# shellcheck disable=SC1091
+[ -r conf ] && . ./conf
+XENSTORED="${XENSTORED:-xenstored}"
+case "$(basename "${XENSTORED}")" in
```

Comment:
Avoid the sub shell, `${XENSTORED##*/}` works in place of base name.

Better yet, why does this case exist? Just drop the whole test, you already provide a mechanism for people to add `--verbose` In `$XENSTORED_ARGS` if they care.

Reply:
I had to look it up, it's picking between two different xenstored binaries, both of which are shipped with the package and I believe take different slightly different options.

Reply:
Yes, but the only difference in the two cases is a default `--verbose` in one. Unless that is a required argument for `oxenstored` (which would shock me), it seems the case as used is superfluous.

Reply:
Good point. I made it as far as reading the documentation that noted that there were different binaries but didn't manage to find the option reference. I can't say I looked particularly hard so it could be that it's as minor a difference as that.